Assets/SuperMultiplayerShooter/Scripts/SampleMainMenu.cs

```diff
@@ -163,123 +302,198 @@ void Update()
         }
 
         /// <summary>
-        /// Called whenever the scene or game ends. Unsubscribe from all channels and channel groups to trigger events immediately.
+        /// Called whenever the scene or game ends. Unsubscribe event listeners.
         /// </summary>
         void OnDestroy()
         {
-            _pubnub.UnsubscribeAll()
-                .Async((result, status) => {
-                if (status.Error)
+            listener.onMessage -= OnPnMessage;
+            listener.onPresence -= OnPnPresence;
+            listener.onObject -= OnPnObject;
+
+            //Clear out the cached players when changing scenes. The list needs to be updated when returning to the scene in case
+            //there are new players.
+            PNManager.pubnubInstance.CachedPlayers.Clear();
+        }
+
+        /// <summary>
+        /// Get the User Metadata given the UserId.
+        /// </summary>
+        /// <param name="Uuid">UserId of the Player</param>
+        private async void GetUserMetadata(string Uuid)
```

Comment:
Do you have a method I can call that essentially does:
1. If the username is cached, return the cached value
2. If the username is NOT cached, obtain the actual value and return that?

Reply:
There is no method for it, but the closest thing is via the OnPnPresence:
```
 //When user joins, check their UUID in cached players to determine if they are a new player.                 
  if (!PNManager.pubnubInstance.CachedPlayers.ContainsKey(result.Uuid))
  {
      GetUserMetadata(result.Uuid);
  }
```

Would you like me to create a function that does that?

Reply:
Added helper function in [commit 657d070](https://github.com/PubNubDevelopers/unity-multiplayer-shooter/pull/3/commits/657d070c41735530e196769610186fb7f0bff9ff).